receiver/stefreceiver/stef.go

```diff
@@ -0,0 +1,163 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package stefreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/stefreceiver"
+import (
+	"context"
+	"errors"
+	"net"
+
+	stefgrpc "github.com/splunk/stef/go/grpc"
+	"github.com/splunk/stef/go/grpc/stef_proto"
+	"github.com/splunk/stef/go/otel/oteltef"
+	stefpdatametrics "github.com/splunk/stef/go/pdata/metrics"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componentstatus"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/stefreceiver/internal"
+)
+
+type stefReceiver struct {
+	cfg        *Config
+	serverGRPC *grpc.Server
+
+	nextMetricsConsumer consumer.Metrics
+	settings            receiver.Settings
+
+	eg errgroup.Group
+}
+
+// Start runs the STEF gRPC receiver.
+func (r *stefReceiver) Start(_ context.Context, host component.Host) error {
+	var err error
+	if r.serverGRPC, err = r.cfg.ServerConfig.ToServer(
+		context.Background(), host, r.settings.TelemetrySettings,
+	); err != nil {
+		return err
+	}
+
+	r.serverGRPC = grpc.NewServer()
```

Comment:
Here it throwing away the server created from the config, I guess this one is some leftover.

Reply:
That's a big miss. Thank you for catching this.